### PR TITLE
feat: add preorder on magma laws

### PR DESCRIPTION
**Pull request by goens**
This builds on the `MagmaLaw` structure defined in #113 and thus depends on it.

### CODE REVIEW

**Comment by goens** [comment]
awaiting-review

**Comment by goens** [comment]
This seems to conflict with #123 (which also defines `MagmaLaw`), can also refactor it to use those definitions once that's merged.

**Comment by teorth** [comment]
It seems that #123 is merged now, so what is the status on compatibility?

**Comment by codyroux** [comment]
It might be worth defining `l ≤ l'` as `{l} ⊧ l'` while we're at it, but it's not critical.

**Comment by goens** [comment]
good idea @codyroux, thanks, I just added that as an equivalent characterization (`l₁ ≤ l₂ ↔ {l₁} ⊧ l₂`), thought the other definition was simpler without having to deal with the singleton

It's integrated with #132 now, I'm happy to merge this if you're happy (@teorth)

File: equational_theories/FreeMagma.lean
Comment (madvorak) [inline equational_theories/FreeMagma.lean:22]: I would call it `instance FreeMagma.isMagma` instead.
Then you don't need the `_root_` thing.

File: equational_theories/Completeness.lean
Comment (madvorak) [inline equational_theories/Completeness.lean:29]: double space

File: equational_theories/Preorder.lean
Comment (madvorak) [inline equational_theories/Preorder.lean:21]: This should work afaik:
```
theorem implies_refl (l : MagmaLaw α) : l ≤ l :=
  fun _ _ => rfl
```

File: equational_theories/FreeMagma.lean
Comment (madvorak) [inline equational_theories/FreeMagma.lean:37]: When we are at it, colon goes at the end of the previous line.

File: equational_theories/Preorder.lean
Comment (goens) [reply equational_theories/Preorder.lean:21]: nope, it doesn't:
```
type mismatch
  rfl
has type
  ?m.104 = ?m.104 : Prop
but is expected to have type
  satisfiesPhi x✝ l : Prop
  ```
  but even if it did, what's wrong with the current proof? looks perfectly reasonable to me.

**Comment by goens** [comment]
thanks for the review @madvorak, I've made all the style changes you suggested (except the proof change that didn't work)

File: equational_theories/FreeMagma.lean
Comment (madvorak) [reply equational_theories/FreeMagma.lean]: Sorry for the misleading comment.

File: equational_theories/Preorder.lean
Comment (madvorak) [reply equational_theories/Preorder.lean:21]: Sorry for the misleading comment.

**Review by madvorak — APPROVED** [review]

File: equational_theories/FreeMagma.lean
Comment (codyroux) [reply equational_theories/FreeMagma.lean]: That's better for sure!

**Comment by pitmonticone** [comment]
Please let me know if there are any further review comments @codyroux. If there are not, I will merge it so we can merge #165 too.

**Comment by pitmonticone** [comment]
Thanks everyone! Merged.

**Pull request merged**
